hw/dv/dpi/common/tcp_server/tcp_server.c

```diff
@@ -351,6 +351,13 @@ static void *server_create(void *ctx_void) {
     rv = select(mfd + 1, &read_fds, NULL, NULL, &timeout);
 
     if (rv < 0) {
+      if (errno == EINTR) {
+        // On interrupt we want to retry; according to the man page
+        // the timeout is now undefined.
+        timeout.tv_sec = 0;
```

Comment:
I'm not completely sure about this bit. If I understand the man page correctly, it is just saying that the function might have trashed timeout (see the section called "The timeout").

So I don't think we need to do this change, *but* we should probably set all of `timeout` every time, so just need to move the write to `tv_sec` inside the loop.

Reply:
Duh. I've just re-read your change and now I understand what you're doing! I think it would still be much simpler to move this line to appear just before the write to `tv_usec`. I think the existing code is a bit silly (because `timeout.tv_sec` might get trashed even if the `select` call succeeds), but the fix is very easy.

Reply:
Thanks for the feedback, I pushed an update.

Reply:
(oops, pushed again to remove the now-extraneous initial init of `tv_sec`).